insonmnia/worker/network/manager.go

```diff
@@ -0,0 +1,292 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/sonm-io/core/insonmnia/worker/network/tc"
+	"github.com/sonm-io/core/util/multierror"
+	"go.uber.org/zap"
+)
+
+const (
+	networkPrefix       = "sonm"
+	networkIfbPrefix    = "ifb-"
+	driverBridge        = "bridge"
+	tagSonmNetwork      = "com.sonm.network"
+	tagSonmNetworkAlias = "com.sonm.network.alias"
+	maxLinkNameLen      = 12
+)
+
+// Action is an abstraction of some action that can be rolled back.
+type Action interface {
+	// Execute executes this action, returning error if any.
+	Execute(ctx context.Context) error
+	// Rollback rollbacks this action, returning error if any.
+	Rollback() error
+}
+
+// ActionQueue represents a queue of executable actions.
+// Any action that fails triggers cascade previous actions rollback.
+type ActionQueue struct {
+	vec []Action
+	mu  sync.Mutex
+}
+
+func NewActionQueue(actions ...Action) *ActionQueue {
```

Comment:
I understand that it it for rollback, but it's quite strange, because those actions never been executed.

Reply:
Agree.